R/tab_create_modify.R

```diff
@@ -162,6 +233,93 @@ tab_spanner <- function(data,
   data
 }
 
+resolve_spanner_level <- function(
+  data,
+  column_names,
+  spanner_ids,
+  level
+) {
+
+  any_existing_spanners <- dt_spanners_exists(data = data)
+
+  # If explicitly providing a `level`, ensure it's at least 1 or above
+  if (!is.null(level)) {
+
+    if (level < 1) {
+      stop(
+        "A spanner level of ", level, " cannot be set:\n",
+        "* Please choose a `level` value greater than or equal to `1`",
+        call. = FALSE
+      )
+    }
+
+    return(as.integer(level))
+  }
+
+  # If spanning exclusively over column labels, and there aren't any
+  # existing spanners, then the new spanner level is always `1`
+  if (length(spanner_ids) < 1 && !any_existing_spanners) {
+    return(1L)
+  }
+
+  spanners_existing <- dt_spanners_get(data = data)
+
+  if (length(spanner_ids) < 1 && any_existing_spanners) {
+
+    highest_level <- 0L
+
+    spanners_existing <-
+      spanners_existing %>%
+      dplyr::select(spanner_id, vars, spanner_level)
+
+    for (i in seq_len(nrow(spanners_existing))) {
+
+      spanned_vars <- unlist(spanners_existing[[i, "vars"]])
+      level <- spanners_existing[[i, "spanner_level"]]
+
+      if (any(column_names %in% spanned_vars)) {
+        if (level > highest_level) {
+          highest_level <- level
+        }
+      }
+    }
+
+    return(highest_level + 1L)
+  }
+
+  # If spanning over any existing spanners then the new spanner
+  # level is one greater than the highest of the spanners with `spanner_ids`
+  spanner_levels_under <-
+    spanners_existing[["spanner_level"]][
+      spanners_existing[["spanner_id"]] %in% spanner_ids
+    ]
+
+  max(spanner_levels_under) + 1L
+}
+
+resolve_spanned_column_names <- function(
+  data,
+  column_names,
+  spanner_ids
+) {
+
+  if (length(spanner_ids) > 0) {
+
+    spanners_existing <- dt_spanners_get(data = data)
+
+    column_names_associated <-
+      unlist(
+        spanners_existing[["vars"]][
+          spanners_existing[["spanner_id"]] %in% spanner_ids
+        ]
```

Comment:
```suggestion
        spanners_existing[spanners_existing[["spanner_id"]] %in% spanner_ids][["vars"]]
```

Reply:
Made this change.

Reply:
This actually results in a subsetting error. Going to revert to previous statement for now.